File: HousingAPI2/HousingAPI/Dtos/HouseDto.cs
namespace HousingAPI.Dtos
{
    public class HouseDto
    {

        public int Id { get; set; }
        public int Number { get; set; }
        public string? Street { get; set; }
        public string? City { get; set; }
        public string? Country { get; set; }
        public string? PostalCode { get; set; }
        public ICollection<ApartmentDto>? Apartments { get; set; }
    }
}

File: housing-client/src/app/models/house-dto.ts


export interface HouseDto {
  id: number;
  number: number;
  street: string;
  city: string;
  country: string;
  postalCode: string;
  apartments: ApartmentDto[];
}

export interface ApartmentDto {
  id: number;
  number: number;
  floor: number;
  roomCount: number;
  population: number;
  totalArea: number;
  livingArea: number;
  houseId: number;
  houseLink: string;
  residents: ResidentDto[];
}

export interface ResidentDto {
  id: number;
  firstName: string;
  lastName: string;
  personalCode: string;
  dateOfBirth: Date;
  phone: string;
  email: string;
  password: string;
  apartmentId: number;
  apartmentLink: string;
  apartmentNumber?: number;
  isOwner: boolean;
  userId: string;
}
